components/style/stylist.rs

```diff
@@ -539,13 +534,13 @@ impl Stylist {
 
                     // Don't let a prefixed keyframes animation override a non-prefixed one.
                     let needs_insertion = keyframes_rule.vendor_prefix.is_none() ||
-                        self.animations.get(keyframes_rule.name.as_atom()).map_or(true, |rule|
+                        origin_cascade_data.animations.get(keyframes_rule.name.as_atom()).map_or(true, |rule|
```

Comment:
While you're here, the indentation here is a bit weird, maybe break it up a bit, or add braces around the closure body? (If you feel like it)